src/sysfiles.c

```diff
@@ -1654,6 +1654,17 @@ Obj OnCharReadHookOutFuncs = 0;/* a list of GAP functions with 0 args */
 Obj OnCharReadHookExcFds = 0;  /* a list of UNIX file descriptors */
 Obj OnCharReadHookExcFuncs = 0;/* a list of GAP functions with 0 args */
 
+// In HPC-GAP we force this to always be active, so we can make use of
+// FreezeStdin to stop HPC-GAP reading stdin while forked processes are running
+Int OnCharReadHookActiveCheck(void)
+{
+#ifdef HPCGAP
+    return 1;
+#else
+    return OnCharReadHookActive != 0;
+#endif
```

Comment:
I am sorry, but I still don't understand what this is doing. What does this hook have to do with HPC-GAP?

As far as I can tell, this hook normally is *never* active in GAP, unless somebody uses `InstallCharReadHookFunc`. But nothing in GAP itself does; among packages, I only see it being used in `sgpviz`.

So my guess is that this is done because `HandleCharReadHook` uses `select` and somehow this resolves the issue at hand "by accident". But isn't that a rather hackish way to achieve this goal?

Also, if this is safe, why don't we just always enable this hook? Conversely, what happens if people try to use `InstallCharReadHookFunc` -- will this interfere somehow? (My guess is "no", but I don't understand what's going on, so I don't tryst my guess).

Reply:
I have added `FreezeStdin`, a global variable which, when set, stops GAP from reading from stdin. The code for this is in the char read hook code, because it requires the use of select.

I have a better way of writing this, I'm going to change it to `OnCharReadHookActive != 0 || FreezeStdin`, which should make more obvious when I'm using this function.

The reason I don't just turn this on always is, looking in the history, it was always turned on at one point then turned off, as some people's systems reported bad behaviour with readline.

Reply:
Ahh, I somehow totally missed that the `FreezeStdin` checks are in `HandleCharReadHook` (entirely my fault, of course). Nevertheless, I still find it strange (but not more so than various other things we do in the kernel, so, whatever :-). I don't mind merging this PR as-is then, although a check `OnCharReadHookActive != 0 || FreezeStdin` indeed would be helpful, IMHO.

But as I understand it, you are planning a revision anyway?